### DISCUSSION TITLE
How to run a workflow step in a different namespace

### DISCUSSION THREAD

**Discussion by chandanpjain** [argoproj/argo-workflows#7655] [Q&A] [answered]
Our usecase is to have a workflow to fetch some files from a pod running in a user defined namespace.
To achieve this, we have a webhook eventsource configured which will trigger the workflow.

Our workflow steps:
1. Verify if the requested namespace exists (obtained from webhook request body).
2. If present, query the pod in the target namespace and get to know which node the pod is running (This is required as the persistentvolume we are using is RWO and the additional pod we are planning to run should be on the same node to mount the volume).
3. Once we know the node, run another pod in the same node, mount the volume of pod from step 2 to the new pod which will enable us to get the files requried and upload (to S3).

To get step 3 working, we will have to run a pod in a different namespace than the ones on which first 2 steps are running or we have to run all the steps in the target namespace. If we have to do this, we might have to create a serviceaccount and roles in every namespace to ensure the workflow can be triggered using the SA. We are looking to avoid this and instead figure out a way to execute only a single step in the target namespace.

Could we get some help in understanding if executing a step in a different namespace is even possible or if there is any other suggested way to achieve this ?

**Comment by terrytangyuan** [comment, marked as answer]
Yes, you can use the cluster install manifest to install Argo Workflows: https://github.com/argoproj/argo-workflows/blob/master/manifests/install.yaml

Once installed, you'll have to play with K8s RBAC to provide the necessary permissions for the controller to manage resources in other namespaces.
